### ISSUE TITLE
Test scenarios

### ISSUE THREAD

**Issue by xsebek** [swarm-game/swarm#388]
Currently, we check that scenarios can be parsed.

I propose we test that they can be solved:

- add a `solution` field to scenarios, which will contain the winning program
- load this program to `base`
- run CESK machine
- verify the `win` condition

This would help us make sure that scenarios are functional, which will be harder once we have many of them for the tutorial.

It would also create an easy way to write tests without knowing any Haskell!

**Comment by byorgey** [comment]
I love this idea!

I do think we will have to make it optional, though.  Especially for scenarios that use a random world seed, it may be very difficult or impossible to write a single program which will always work.  Maybe we could specify a particular seed along with the solution.  In any case, even if it's optional, having and testing solutions for as many scenarios as we can is still a big win.

#378 is related and maybe it makes sense to do these at the same time. In the context of having a sequence of objectives I guess we could have a separate solution for each.

**Comment by xsebek** [comment]
Sure, the field can be optional (for creative it pretty much has to be) and then nothing is tested.

I must admit, I did not consider how multiple objectives will interact with this, but I hope it will be easy to generalize.

For a POC we have the ability to `initGameState` with scenario and program from `.sw` so I started a WIP at [wip/test-scenario](https://github.com/swarm-game/swarm/tree/wip/test-scenario).
However, I am hopelessly clueless, as to how I should check the win condition.
